h2/src/docsrc/help/help.csv

```diff
@@ -1168,6 +1168,34 @@ This setting can be appended to the database URL: ""jdbc:h2:test;CACHE_SIZE=8192
 SET CACHE_SIZE 8192
 "
 
+"Commands (Other)","SET COLUMN_NAMING_RULES","
+SET COLUMN_NAMING_RULES = { DEFAULT | EMULATE=... | MAX_IDENTIFIER_LENGTH=... | REGULAR_EXPRESSION_MATCH_ALLOWED=... | REGULAR_EXPRESSION_MATCH_DISALLOWED=... | DEFAULT_COLUMN_NAME_PATTERN=... }
+","
+This setting controls the rules dealing with how column names are generated  
+in select statements.
+
+Administrator rights are required to execute this command, as it affects all 
+connections. This setting is not persistent.
+Emulation settings can be appended to the database URL: 
+""jdbc:h2:test;COLUMN_NAME_RULES=EMULATE=ORACLE30""
+
+It is not recommended to use overly restrictive settings as system 
+columns can accidentally get their identifiers truncated leading to 
+undefined behavior. A minimum value of 30 is enforced.
+","
+-- Emulate popular database settings
+SET COLUMN_NAME_RULES=EMULATE=ORACLE128 -- Emulate Oracle 128 character column identifiers
+SET COLUMN_NAME_RULES=EMULATE=ORACLE30; -- Emulate Oracle 30 character column identifiers
+SET COLUMN_NAME_RULES=EMULATE=POSTGRES; -- Emulate Postgres
+SET COLUMN_NAME_RULES=DEFAULT; -- Reset back to default H2 allow-anything rules
+-- Custom settings
+SET COLUMN_NAME_RULES=MAX_IDENTIFIER_LENGTH = 30; -- Custom configure max identifier length (minimum value is 30)
+SET COLUMN_NAME_RULES=REGULAR_EXPRESSION_MATCH_ALLOWED = '[A-Za-z0-9_]+';  -- Custom configure Java RegEx for allowed characters
+SET COLUMN_NAME_RULES=REGULAR_EXPRESSION_MATCH_DISALLOWED = '[^A-Za-z0-9_]+'; -- Custom configure Java RegEx for disallowed characters
+SET COLUMN_NAME_RULES=DEFAULT_COLUMN_NAME_PATTERN = '_unnamed$$'; -- Custom configure default column name, $$ is replaced by column Nth ordinal
+SET COLUMN_NAME_RULES=GENERATE_UNIQUE_COLUMN_NAMES = 1; -- Generate unique column names (default is 0, meaning false)
+"
+
```

Comment:
hmmm, I'm not a big fan of unnecessary flexibility. I'd rather just hardcode a couple of rules for the different database styles we support

but maybe there is some use case I am missing?

Reply:
h2 does a really bad job at naming columns with any rules or standards (in fact it makes no effort to do that whatsoever, other than basic SQL alias and SQL expression capture) . I'm used to seeing pretty, generated column names from Oracle and SQL Server. This bug fix scernario included a column name and more then 30 charactes column name - with a CR/LF in it - so I'm trying to regain that nirvana of other DBs. I don't know much about other DB's but they have more rules and capabilities - and that's what I'm trying to capture. 

Whats missing is toUpper() or toLower() support - and column name clash detection (I think Oracle enforces unique names or errors otherwise). If anything there's more to do here than I have had time to do.